README.md

```diff
@@ -1,127 +1,130 @@
 
-  if (needsUpdate) {
+  if (newTodos !== undefined && newTodos !== _todos) {
+    _todos = Object.freeze(newTodos);
     TodoStore.emitChange();
   }
-
 });
 ```
 
 Use `Dispatcher.waitFor` if you need to ensure handlers from other stores run first.
 
-```javascript
-var mcFly = new McFly();
-var Dispatcher = mcFly.dispatcher;
-var OtherStore = require('../stores/OtherStore');
-var _todos = [];
-
-function addTodo(text, someValue) {
-  _todos.push({ text: text, someValue: someValue });
+```js
+case 'ADD_ANOTHER_TODO': {
+  flaxs.dispatcher.waitFor([OtherStore.dispatcherID]);
+  const someValue = OtherStore.getSomeValue();
+  if (!_todos.includes(someValue)) {
+    newTodos = addTodo(_todos, someValue);
+  }
+  break;
 }
-
- ...
-
-    case 'ADD_TODO':
-      Dispatcher.waitFor([OtherStore.dispatcherID]);
-      var someValue = OtherStore.getSomeValue();
-      addTodo(payload.text, someValue);
-      break;
-
- ...
 ```
 
 Stores are also created a with a ReactJS component mixin that adds and removes store listeners that call a **storeDidChange** component method.
 
+### React Integration
+
 Adding Store eventing to your component is as easy as:
 
 ```javascript
-var TodoStore = require('../stores/TodoStore');
+const { TodoStore } from '../stores/TodoStore';
 
 var TodoApp = React.createClass({
 
   mixins: [TodoStore.mixin],
 
+  render() {}
 })
 ```
 
-## Reducers
+**Important**.  The mixin component created in the store, is preserved to keep compatibility with McFly as well, but will get deprecated once `react-flaxs` project sees the light.  Flaxs will not contain itself any dependency from ReactJS.
 
-Motivated by Redux, the most popular flux implementation, reducers are reduced dispatched callbacks that modify a single state of the application.  Similar to Stores in mcFly but suppressing the getters object and incorporating by default the global state of the master store.
-
-Reducers in this project follow the 3 basic principles of Redux ([Read about Redux](http://redux.js.org/docs/introduction/ThreePrinciples.html)), which are:
-
-- **A single master store**. A single store containing no getters and a unique state for the whole application.
-- **Immutable state**. That means that the store contains a state which is a one-level object with immutable attribute values. The only way to mutate the state is to emit an action, an object describing what happened.
-- **State changes are made via pure functions**. Reducers are just pure functions that take the previous state and an action, and return the next state. Many reducers can make modifications through different portions of the state in different reducers, so we avoid chaining actions in our components and we avoid triggering actions inside our stores/reducers.
+## Reducers
 
-Unlike redux, mcfly uses synchrously EventEmitter to make changes in a store.
+Unlike redux, flaxs uses asynchronously an EventEmitter to make changes in a store.  This was part of the heritage from mcFly project.
```

Comment:
It is a very good point, nonetheless the principle of flux is to leave it asynchronous.  SInce `flaxs` is a tool that works well as transition between `flux` => `redux`, We could evaluate then the use of `redux` with `flaxs`.
I believe the whole point of doing them asynchronous was to leverage the `EventEmitter` publish/subscribe pattern.  Which not sure as well, if we change it to synchronous, how will behave for backwards compatibility.

Definitively, I am looking forward to do it, tho.